cmd/sentinel/cmd/sentinel.go

```diff
@@ -1818,7 +1818,8 @@ func (s *Sentinel) Start(ctx context.Context) {
 			return
 		case <-timerCh:
 			go func() {
-				s.clusterSentinelCheck(ctx)
+				for !s.clusterSentinelCheck(ctx) {
+				}
```

Comment:
I'm not very familiar with the existing Stolon code, so this may be a dumb question...

if I'm following this loop correctly, it initializes a timer with an duration of 0, basically "fire as often as you can"

and in the original code, when the timer fired, it called `clusterSentinelCheck` once, and when that returned it wrote a value to `endCh`, which causes the timer to be replaced with one that runs every `s.sleepInterval`

with this change, `clusterSentinelCheck` gets called in a loop, so this anonymous goroutine will run for longer than it did previously

and that 0-duration timer will still be running in the background, so it seems like we'd be spawning a potentially unlimited number of goroutines here?

Reply:
Oh I think each `Timer` fires only once, so effectively the containing loop means "call `clusterSentinelCheck()` every `s.sleepInterval`" (with the first iteration firing immediately). So the second goroutine won't be created until the first one ends with pushing to `endCh`. I don't know if this is the convention to make repeated calls with intervals in Go, but Stolon is doing so in the keepers code too (that one probably makes more sense because it multiplexes several repeated calls in one outer loop).

Reply:
nevermind, I mixed up `time.Timer` and `time.Ticker`